### ISSUE TITLE
Module request: FOR-A HVS 390

### ISSUE THREAD

**Issue by christianskov** [bitfocus/companion-module-requests#29]
The ability to control the **FOR-A HVS 390** via the Editor RS-422 port. 
You need to buy the EDITOR option as a firmware upgrade at FORA. 

You would also need a generic Ethernet to RS422 converter. 

Attached as PDF:  
The manual descriping how to turn on the **Editor function** and 
**EDITOR protocol command list** 

Use the Companion to control the HVS390 mixer, aux, ME crospoints, recall memory ect. 

[HVS39ED-E-V1.pdf](https://github.com/bitfocus/companion/files/3084150/HVS39ED-E-V1.pdf)
[HVS390-Mixer and Editor.pdf](https://github.com/bitfocus/companion/files/3084151/HVS390-Mixer.and.Editor.pdf)

Thanks for a great tool!
Christian

**Comment by Fader-Monkey** [comment]
Michael Allen has a for-a module going for the HVS series. right now we have tested it with the HVS 100 and we are currently working through adding the HVS 2000 (it has a different command structure.) I wonder if the 390 shares its protocol with either switcher. 

we went with the web socket approach instead of the DVS route because its free and does not require any extra hardware (the MU sub port just needs to be on a network)

you can see what we have done with the 100/200 here: https://github.com/bitfocus/companion-module-fora-hvs/issues/1

if you want to grab a network log of the web panel we can see if it shares the same commands.

**Comment by christianskov** [comment]
Hi, thats great news. 
I will try to fetch the. Information on the 390. 
In the mean time we did a node-api to websocket application, enabling us to sent http / get request to the mixer via web socket. 
We are still interested in a more stable approach, it seams the the mixer only supports one web socket connection at a time, it seems unstable in loading the webpage, when our “middleware” application is talking over the web socket. 
Christian

**Comment by Fader-Monkey** [comment]
yes one connection at a time is a recurring theme with for-a. either routers or switchers. from experience I have had a good track record with the HVS module staying connected (it skips the login on the web pages and connects directly to the websocket) there is also a reconnect command that is helpful if it runs into a glitch.

**Comment by josephdadams** [comment]
This is now available as the **`fora-hvs`** module (latest version v1.2.6).

You can install it from Companion's module store, or see https://bitfocus.io/connections/fora-hvs

Closing as implemented — please open a new issue on [the module's repository](https://github.com/bitfocus/companion-module-fora-hvs/issues) if your specific model still doesn't work.
